Cabal/Distribution/Simple/Program/HcPkg.hs

```diff
@@ -234,7 +234,8 @@ dump hpi verbosity packagedb = do
 
   output <- getProgramInvocationOutput verbosity
               (dumpInvocation hpi verbosity packagedb)
-    `catchIO` \_ -> die $ programId (hcPkgProgram hpi) ++ " dump failed"
+    `catchIO` \e -> die $ programId (hcPkgProgram hpi) ++ " dump failed: "
+                       ++ show e
```

Comment:
I believe that with the advent of `displayException`, `show` now shows the derived form for things like IOExceptions, which means we probably ought to use `displayException`, but doing so requires CPP. Sigh.